test/unit/uk/gov/hmrc/apinotificationqueue/service/WarningEmailPollingServiceSpec.scala

```diff
@@ -0,0 +1,95 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.apinotificationqueue.service
+
+import akka.actor.ActorSystem
+import org.joda.time.DateTime
+import org.mockito.ArgumentCaptor
+import org.mockito.ArgumentMatchers.{any, eq => ameq}
+import org.mockito.Mockito._
+import org.scalatest.concurrent.Eventually
+import org.scalatest.mockito.MockitoSugar
+import uk.gov.hmrc.apinotificationqueue.config.ServiceConfiguration
+import uk.gov.hmrc.apinotificationqueue.connector.EmailConnector
+import uk.gov.hmrc.apinotificationqueue.model.{Email, SendEmailRequest}
+import uk.gov.hmrc.apinotificationqueue.repository.{ClientOverThreshold, NotificationRepository}
+import uk.gov.hmrc.http.HttpResponse
+import uk.gov.hmrc.play.test.UnitSpec
+
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.concurrent.Future
+
+class WarningEmailPollingServiceSpec extends UnitSpec with MockitoSugar with Eventually {
+
+  trait Setup {
+
+    val mockNotificationRepository = mock[NotificationRepository]
+    val mockEmailConnector = mock[EmailConnector]
+    val mockServiceConfiguration = mock[ServiceConfiguration]
+    val testActorSystem = ActorSystem("WarningEmailPollingService")
+
+    val year = 2017
+    val monthOfYear = 7
+    val dayOfMonth = 4
+    val hourOfDay = 13
+    val minuteOfHour = 45
+    val timeReceived = new DateTime(year, monthOfYear, dayOfMonth, hourOfDay, minuteOfHour)
+    val latestReceived = timeReceived.plus(1)
+    val clientId1 = "clientId1"
+    val clientOverThreshold1 = ClientOverThreshold(clientId1, 2, timeReceived, latestReceived)
+
+    val sendEmailRequest = SendEmailRequest(List(Email("some-email@address.com")),
+      "customs_pull_notifications_warning",
+      Map("clientId_0" -> "some-client-id",
+        "notificationTotal_0" -> "some-total",
+        "oldestNotification_0" -> "some-date",
+        "latestNotification_0" -> "some-date",
+        "queueThreshold" -> "some-threshold"),
+      force = false)
+  }
+
+  "WarningEmailService" should {
+    "send an email" in new Setup {
+      when(mockServiceConfiguration.getInt("notification.email.queueThreshold")).thenReturn(2)
+      when(mockServiceConfiguration.getString("notification.email.address")).thenReturn("some-email@address.com")
+      when(mockNotificationRepository.fetchOverThreshold(2)).thenReturn(Future.successful(List(clientOverThreshold1)))
+      when(mockServiceConfiguration.getInt("notification.email.interval")).thenReturn(1)
+
+      val warningEmailService = new WarningEmailPollingService(mockNotificationRepository, mockEmailConnector, testActorSystem, mockServiceConfiguration)
+      Thread.sleep(1000)
```

Comment:
can we use `eventually` for verification section rather than `Thread.sleep` ?

Reply:
Done